bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -749,13 +770,27 @@ public synchronized ModuleHandlerFactory addingService(ServiceReference/* <Modul
             }
         }
         if (notInitailizedRules != null) {
-            for (String rUID : notInitailizedRules) {
-                setRule(rUID);
+            for (final String rUID : notInitailizedRules) {
+                scheduleRuleInitialization(rUID);
             }
         }
         return mhf;
     }
 
+    private synchronized void scheduleRuleInitialization(final String rUID) {
+        Future f = scheduleTasks.get(rUID);
+        if (f == null) {
+            ScheduledExecutorService ex = getScheduledExecutor();
+            f = ex.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    setRule(rUID);
+                }
+            }, 500, TimeUnit.MILLISECONDS);
+            scheduleTasks.put(rUID, f);
+        }
```

Comment:
I think no, the task will be executed 500 ms after fist try. I don't see a reason to be 500 ms after the last one.